cpp/src/strings/json/json_path.cu

```diff
@@ -952,8 +951,12 @@ __launch_bounds__(block_size) __global__
   if (out_valid_count) {
     size_type block_valid_count =
       cudf::detail::single_lane_block_sum_reduce<block_size, 0>(warp_valid_count);
-    if (threadIdx.x == 0) { atomicAdd(out_valid_count.value(), block_valid_count); }
+    if (threadIdx.x == 0) {
+      if (out_valid_count.has_value()) { *(out_valid_count.value()) = 0; }
+      atomicAdd(out_valid_count.value(), block_valid_count);
+    }
```

Comment:
Seems wrong to me (please correct if I'm also wrong). Should be this?
```suggestion
    if (tid == 0) { *(out_valid_count.value()) = 0; }
    if (threadIdx.x == 0) { atomicAdd(out_valid_count.value(), block_valid_count); }
```

Reply:
Oh no, the solution above is wrong.

Reply:
Yeah it is, we're reverting it.

Reply:
I'm so sorry, I just carefully think about this and have to admit that we don't have a better way to initialize the variable inside the kernel. We have to initialize it once (by the first thread, `tid == 0`) but are not guaranteed to have that thread will always execute first.

So we have to initialize at the caller anyway.

Reply:
@trxcllnt suggested that I revert, since the above actually probably just reintroduces the bug, so I'm going to do that! :)